### ISSUE TITLE
Possible EventEmitter memory leak

### ISSUE THREAD

**Issue by leonrinkel** [leonrinkel/reimagined-umbrella#9]
```
(node:18732) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 subscriptions listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18732) UnhandledPromiseRejectionWarning: undefined
(node:18732) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 6)
(node:18732) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:18732) UnhandledPromiseRejectionWarning: undefined
(node:18732) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 7)
```

**Comment by leonrinkel** [comment]
```
info: opening influx write api
{ influxUrl: 'http://172.17.0.1:8086',
  influxOrg: 'reimagined-umbrella',
  influxBucket: 'reimagined-umbrella',
  influxFlushInterval: 10000 }
info: successfully connected to coinbase
info: successfully subscribed to heartbeat channels
info: successfully subscribed to ticker channels
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
warn: received no heartbeats for more than 10000 ms, trying to reconnect...
warn: received no heartbeats for more than 10000 ms, trying to reconnect...
warn: received no heartbeats for more than 10000 ms, trying to reconnect...
info: successfully reconnected
info: reconnected, trying to resubscribe...
error: resubscription failed, closing socket...
info: successfully reconnected
info: reconnected, trying to resubscribe...
error: resubscription failed, closing socket...
info: successfully reconnected
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
warn: received no heartbeats for more than 10000 ms, trying to reconnect...
info: successfully reconnected
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: successfully resubscribed
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
info: reconnected, trying to resubscribe...
...
(node:1) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 subscriptions listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
...
```

**Comment by leonrinkel** [comment]
Resolved by #19
